src/Resources/contao/widgets/PickerWidget.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao;
+
+
+/**
+ * Provide methods to handle input field "picker".
+ *
+ * @property boolean        $multiple
+ * @property array          $rootNodes
+ * @property string         $fieldType
+ * @property string         $foreignTable
+ * @property string         $context
+ * @property string         $labelField
+ * @property string         $icon
+ * @property boolean        $sortable
+ * @property array|callable $label_callback
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+class PickerWidget extends \Widget
+{
+
+	/**
+	 * Submit user input
+	 * @var boolean
+	 */
+	protected $blnSubmitInput = true;
+
+	/**
+	 * Template
+	 * @var string
+	 */
+	protected $strTemplate = 'be_widget';
+
+
+	/**
+	 * Load the database object
+	 *
+	 * @param array $arrAttributes
+	 */
+	public function __construct($arrAttributes=null)
+	{
+		$this->import('Database');
+		parent::__construct($arrAttributes);
+	}
+
+
+	/**
+	 * Return an array if the "multiple" attribute is set
+	 *
+	 * @param mixed $varInput
+	 *
+	 * @return mixed
+	 */
+	protected function validator($varInput)
+	{
+		$this->checkValue($varInput);
+
+		if ($this->hasErrors())
+		{
+			return '';
+		}
+
+		// Return the value as usual
+		if ($varInput == '')
+		{
+			if ($this->mandatory)
+			{
+				$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['mandatory'], $this->strLabel));
+			}
+
+			return '';
+		}
+		elseif (strpos($varInput, ',') === false)
+		{
+			return $this->multiple ? array(intval($varInput)) : intval($varInput);
+		}
+		else
+		{
+			$arrValue = array_map('intval', array_filter(explode(',', $varInput)));
+
+			return $this->multiple ? $arrValue : $arrValue[0];
+		}
```

Comment:
we had problems on saving multiple selected ids and afterwards search them via like in DB - so do we have to cast to string at this point?

```        
        if ($this->multiple)
        {
            // Cast to string to search in serialized array via db
            if (!empty($this->varValue)) {
                foreach ($this->varValue as &$value) {
                    $value = (string)$value;
                }
            }
        }
```

Reply:
I have never encountered such problem but if that's the case then also other widgets are affected. What query have you used to search the serialized data? Or did you set the field to handle data in `csv` format?

Reply:
We use something like this ` WHERE (xx LIKE '%"1"%' OR xx LIKE '%"10"%') ` - it's similar to the backend-filter for members.

Yes, i guess it's also an problem for other widgets. But at the moment I don't know if there is an integer-picker in the core. the file picker uses uuid, so it's string...

Reply:
https://github.com/contao/core-bundle/blob/master/src/Resources/contao/widgets/PageTree.php#L126

Reply:
I know that code - because I also copied that initialy for our own picker ;-)
it saves the ids correctly but you won't be able to search via LIKE...

Reply:
Imho that's not related to this pull request…